iotfunctions/pipeline.py

```diff
@@ -1122,31 +1122,36 @@ def execute_stage(self, stage, df, start_ts, end_ts):
         # entities to retrieve data from
         entities = self.exec_payload_method(method_name='get_entity_filter', default_output=None, raise_error=False)
         usage = 0
+        kpi_function_name = "unknown"
 
         # There are two possible signatures for the execute method
         try:
             result = stage.execute(df=df, start_ts=start_ts, end_ts=end_ts, entities=entities)
 
             usage = self.get_stage_param(stage, 'usage_', usage)
 
+            kpi_function_name = self.get_stage_param(stage, "kpi_function_name", kpi_function_name);
```

Comment:
this line is repeated later. I see a chance to reduce the code by one line :-)

Reply:
Okay, I'll simplify this section